test/ContractTestContext.ts

```diff
@@ -1,4 +1,4 @@
-import { Contract, ContractFactory } from "ethers";
+import { Contract } from "ethers";
```

Comment:
Not for this PR but the changes here are quite confusing to follow offline. Lets schedule some in person time so you can walk us through it. Non blocker for PR